### PR TITLE
Refactor PeerConnection tests

### PR DESCRIPTION
**Pull request by mickel8**
The idea is to have:
* notification tests
* api tests
* other tests

That's why I decided to move some tests. 
Any thoughts on that?

Other than that I also:
* moved most negotiation needed assertions into other, already existing tests
* fixed DTLS cert fingerprint bug
* added test for Firefox SDP offer

### CODE REVIEW

File: test/ex_webrtc/peer_connection_test.exs [hunk 1]
```diff
@@ -36,57 +36,76 @@ defmodule ExWebRTC.PeerConnectionTest do
                  fingerprint: {:sha256, @fingerprint}
                )
 
-  test "negotiation needed" do
```
Comment (mickel8) [inline test/ex_webrtc/peer_connection_test.exs:39]: I merged this test with other already existing tests (e.g/ add_track, remove_track) or created new ones (e.g add_transceiver)

File: test/ex_webrtc/browser_sdp_test.exs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+defmodule ExWebRTC.BrowserSDPTest do
+  use ExUnit.Case, async: true
+
+  alias ExWebRTC.{MediaStreamTrack, PeerConnection, SessionDescription}
+
+  for browser <- ["firefox"] do
+    test "#{browser} SDP offer" do
```
Comment (mickel8) [inline test/ex_webrtc/browser_sdp_test.exs:7]: Just to make sure we can successfuly parse remote SDP offers both from chromium and ff. Thanks to this I also spotted a bug with DTLS cert fingerprint

File: test/ex_webrtc/peer_connection_test.exs [hunk 3]
```diff
@@ -719,18 +760,30 @@ defmodule ExWebRTC.PeerConnectionTest do
     :ok = PeerConnection.close(pc2)
   end
 
-  test "close/1" do
-    {:ok, pc} = PeerConnection.start()
-    {:links, links} = Process.info(pc, :links)
-    assert :ok == PeerConnection.close(pc)
-    assert false == Process.alive?(pc)
-    Enum.each(links, fn link -> assert false == Process.alive?(link) end)
+  test "reject incoming track" do
```
Comment (mickel8) [inline test/ex_webrtc/peer_connection_test.exs:763]: Alternatively we could put this in `describe "set_transceiver_direction"`

**Review by LVala — APPROVED** [review]
👍

**Pull request merged**